pinc/special_colors.inc

```diff
@@ -78,30 +106,19 @@ function echo_special_legend( $projects_where_clause)
             WHERE $projects_where_clause
         ");
 
-    $curr_specs_array = array();
+    $specials_array = load_special_days();
+    $day_array = [];
 
     while ($cs_row = mysqli_fetch_assoc($currspecs_result)) {
-        $curr_specs_array[] = $cs_row['spec'];
-    }
-
-    $specs_result = mysqli_query(DPDatabase::get_connection(), "
-            SELECT spec_code, display_name, color FROM special_days
-            ORDER BY display_name ASC
-        ");
-
-    $common_array = array();
-
-    // Strip results down to just the ones in use
-    while ($sr_row = mysqli_fetch_assoc($specs_result)) {
-        if (in_array($sr_row['spec_code'], $curr_specs_array)) {
-            $common_array[] = $sr_row;
-        }
+        if(isset($specials_array[$cs_row['spec']]))
+            $day_array[] = $specials_array[$cs_row['spec']];
     }
-    $common_array[] = array('color' => "CCFFFF", 'display_name' => "Authors with recent birthdays");
-    $common_array[] = array('color' => "33CCFF", 'display_name' => "Authors with birthdays today");
-    $common_array[] = array('color' => "FFFF66", 'display_name' => "Other Special");
+    // sort day_array by display_name
+    array_multisort(array_column($day_array, 'display_name'), SORT_ASC, $day_array);
+    // now append the hard-coded special day names
```

Comment:
This retains the current behavior of keeping the "common" special days at the end of the legend. To me it makes more sense to have them be alpha-sorted with the rest of them and that change would be easy to make.

Reply:
I'm not even sure if Other Special is used (or should be) but we don't have a foreign key on the tables to ensures that all projects map to existing values in the `special_codes` table. I'm just going to leave it all as-is for now I think.

Reply:
Sounds good.

I have seen Other Special be used. It's to fill in the gap left between the defined special days and the PM-definable birthdays. It, like the PM-definable birthday, is easily abused.